septima-js-data/src/main/java/com/septima/entities/SqlEntities.java

```diff
@@ -609,11 +647,11 @@ private Database databaseOf(final String aDataSourceName) {
         Objects.requireNonNull(aDataSourceName, "aDataSourceName ia required argument");
         return databases.computeIfAbsent(aDataSourceName, dsn -> {
             try {
-                DataSource ds = Database.obtainDataSource(aDataSourceName);
-                Metadata metadata = Metadata.of(ds);
+                DataSource dataSource = Database.obtainDataSource(aDataSourceName);
                 return new Database(
-                        ds,
-                        metadata,
+                        dataSource,
+                        DataSources.getDataSourceSqlDriver(dataSource),
+                        compileEntities ? Metadata.of(dataSource) : null,
```

Comment:
It's little bit mystifying. If we have compiled entities why we need meta data from data source?

Reply:
'Compile' and 'compiled' are different words, aren't they?